### PR TITLE
Add transport_version to node info json

### PR DESCRIPTION
**Pull request by thecoop**
Add a transport version field to the node info JSON.

This is primarily informative, but also needed for tests in #94597

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodesInfoResponse.java [hunk 1]
```diff
@@ -64,6 +64,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field("ip", nodeInfo.getNode().getHostAddress());
 
             builder.field("version", nodeInfo.getVersion());
+            builder.field("transport_version", nodeInfo.getTransportVersion().id);
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodesInfoResponse.java:67]: This is outputting an int directly into JSON here

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java [hunk 2]
```diff
@@ -40,25 +40,32 @@
...
-    private ByteSizeValue totalIndexingBuffer;
+    private final ByteSizeValue totalIndexingBuffer;
 
     public NodeInfo(StreamInput in) throws IOException {
         super(in);
         version = Version.readVersion(in);
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0)) {
+            transportVersion = TransportVersion.readVersion(in);
+        } else {
+            assert version.before(Version.V_8_8_0);
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java:66]: Is it possible to read a >=8.8.0 node info structure from a <8.8.0 ES node?

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/bwc

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java (hunk 2 above)
Comment (rjernst) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java:66]: No this should not be possible, and I don't think the assert is necessary

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**